front/src/modules/recoil-scope/components/RecoilScope.tsx

```diff
@@ -0,0 +1,24 @@
+import { Context, useState } from 'react';
+import { v4 } from 'uuid';
+
+import { RecoilScopeContext } from '../states/RecoilScopeContext';
+
+export function RecoilScope({
+  SpecificContext,
+  children,
+}: {
+  SpecificContext?: Context<string | null>;
```

Comment:
@lucasbordeau Not sure of this `api`, maybe we can just add a `key` prop that is concatenated with the generated `id` and allow us to use nested context.

Reply:
No because useContext will take the closest react context in the hierarchy, we have to specify a specific context for useContext to bypass the hierarchy of nested contexts.

Reply:
I feel like it's not ideal by working with only one context is not easily feasible, so this solution seems like a good tradeoff for me right now.

Reply:
Ok, let's take a look on that later :)